packages/compiler-cli/src/ngtsc/annotations/src/component.ts

```
@@ -407,8 +423,12 @@ export class ComponentDecoratorHandler implements
       }
     }
 
-    let inlineStyles: string[]|null = null;
-    if (component.has('styles')) {
+    // If inline styles were preprocessed use those
+    let inlineStyles = this.preanalyzeStylesCache.get(node) || null;
+    if (inlineStyles !== null) {
+      this.preanalyzeStylesCache.delete(node);
+      styles.push(...inlineStyles);
+    } else if (component.has('styles')) {
       const litStyles = parseFieldArrayValue(component, 'styles', this.evaluator);
```

Not going trough the `ResourceLoader` here means that forgetting to call `loadNgStructureAsync` silently skips all preprocessing, unlike non-inline resources that would error out if `ResourceLoader.load` is unable to synchronously return the resource data. I see two potential designs to align this:

1. Have both `preprocessInline` and `preprocessInlineAsync` (or some other names)
2. Allow `preprocessInline` to return a `string | Promise<string>` and also call it in the sync-path, requiring that no `Promise` is returned. Maybe `ResourceLoaderContext` has to indicate whether it's sync or async; I'm not sure how useful that would be.

---

There are currently no use cases where a transform would be synchronous so I intentionally avoided the extra complexity of the synchronous path.

It should now assert and throw if `loadNgStructureAsync` is not executed.